azure-keyvault/azure/keyvault/custom/key_vault_id.py

```diff
@@ -0,0 +1,238 @@
+# ---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# ---------------------------------------------------------------------------------------------
+
+try:
+    import urllib.parse as parse
+except ImportError:
+    import urlparse as parse # pylint: disable=import-error
+
+from enum import Enum
+
+
+class KeyVaultCollectionType(Enum):
+    keys = 'keys'
+    secrets = 'secrets'
+    certificates = 'certificates'
+    certificate_issuers = 'certificates/issuers'
+
+
+class KeyVaultId(object):
+    """ 
+    An identifier for an Azure Key Vault resource.
+    """
+
+    def __init__(self, collection, vault, name, version):
+        """
+        :param collection: The resource collection type.
+        :type collection: str
+        :param vault: The vault URI.
+        :type vault: str
+        :param name: The resource name.
+        :type name: str
+        :param version: The resource version.
+        :type version: str
+        """
+        self.vault = vault
+        self.name = name
+        self.version = version or ''
+        self.base_id = '{}/{}/{}'.format(vault, collection, name)
+        self.id = '{}/{}'.format(self.base_id, version) if version else self.base_id
```

Comment:
`id` and `base_id` are really derived values.  With this structure (having them as instance variables), manually changing the base_id would not update the id... or you could completely replace either with gibberish.

Consider changing it so that you can only set vault, name and version, and then have `id` and `base_id` as instance methods that return the appropriate string.